tensorflow/python/kernel_tests/pooling_ops_test.py

```diff
@@ -811,12 +811,9 @@ def _CompareMaxPoolingGradBk(self, input_shape, output_shape, ksize, strides,
       self.assertAllCloseAccordingToType(
           cpu_val, gpu_val, half_rtol=0.01, half_atol=0.01)
 
-  def testMaxPoolingWithArgmax(self):
-    # MaxPoolWithArgMax is implemented only on CUDA.
-    if not test.is_gpu_available(cuda_only=True):
-      return
+  def _testMaxPoolingWithArgmax(self, use_gpu):
     tensor_input = [1.0, 1.0, 1.0, 1.0, 0.0, 1.0, 1.0, 1.0, 1.0]
-    with self.test_session(use_gpu=True) as sess:
+    with self.test_session(use_gpu=use_gpu) as sess:
```

Comment:
The preferred way to write tests is to set use_gpu=True. This will fall back to using the CPU if no GPU is available. Most tests are run in both environments, so both paths should get tested.

Reply:
Ok. Removing setting use_gpu explicitly.